### ISSUE TITLE
Xdocs examples should reject violation comments between annotations and declarations

### ISSUE THREAD

**Issue by smita1078** [checkstyle/checkstyle#20382]
Follow-up of #19803.

Background

Currently Input files are validated by checks in `config/checkstyle-input-checks.xml` that reject violation comments placed between annotations/Javadocs and their target declarations.

Examples:

bad

```
@Override
// violation above 'xxx'
void foo() {}
```

good

```
// violation below 'xxx'
@Override
void foo() {}
```

However, equivalent validation is not applied to xdocs examples under `src/xdocs-examples/resources`.

As a result, patterns such as:

```
@VisibleForTesting
// violation below 'must be private'
public String testString = "";
```

can exist in examples and pass validation.

Task
Apply the same validation used for Inputs to xdocs examples.

Expected behavior

Fail validation when marker comments are placed between:
- annotations and annotated declarations
- Javadocs and declarations

Examples should follow the same conventions as Input files.

**Comment by smita1078** [comment]
While testing the `markerCommentBetweenAnnotationAndTarget` / `markerCommentBetweenJavadocAndTarget` checks, I found they're over-flagging: any marker comment placed after an annotation gets caught, even when it's a perfectly valid inline trailing comment.

**Cases:**

1. Should be flagged (comment alone on its own line, between annotation and target):
```java
@Override
 // violation above 'xxx'
void foo() {}
```

2. Currently flagged, but shouldn't be (inline, same line as annotation):
```java
@Override  // violation above 'xxx'
void foo() {}
```

3. Correctly not flagged (comment before the annotation):
```java
 // violation below 'xxx'
@Override
void foo() {}
```

**Why case 2 fails:** the current rule is `//MODIFIERS/SINGLE_LINE_COMMENT[... and preceding-sibling::ANNOTATION]`. XPath here only sees the AST shape  a comment node sitting after an `ANNOTATION` node in `MODIFIERS`  and that shape is identical whether the comment is on the same line as the annotation or on the line below it. Checkstyle's XPath nodes only expose a `@text` attribute (confirmed in `ElementNode.getAttributeValue`, which returns `null` for everything except `"text"`), so there's no way to query line/column position to tell these two cases apart.

@romani

**Comment by romani** [comment]
> Currently flagged, but shouldn't be (inline, same line as annotation):

yes, xpath is far from ideal implementation approach, I am fine to sacrifice "case 2". It is Better then catching this stuff manually during code review.

**Comment by smita1078** [comment]
> yes, xpath is far from ideal implementation approach, I am fine to sacrifice "case 2". It is Better then catching this stuff manually during code review.

Can we swap these two rules from `MatchXpath` to `RegexpMultiline` for example checks,  it actually catches case 1 without losing case 2, and it fits this config file's existing conventions better anyway (it's already full of `RegexpSingleline` rules for similar marker-comment formatting, e.g. `ensureSpaceAfterSpecificSingleLineComment`, `xdocStartComment`).

Instead of asking "is this comment a sibling after an `ANNOTATION` node in the AST" (which can't see line breaks, hence the case 2 problem), ask "is the annotation/javadoc-close alone on its own line, immediately followed by a line that is *only* the marker comment." Regex can see that distinction because it works on raw text with real line breaks, where XPath only sees tree shape.

**Comment by smita1078** [comment]
I can go ahead and create a draft PR to see if this works..

**Comment by romani** [comment]
We can try to see what works better, but usually xpath is slightly better than regexp

**Comment by romani** [comment]
fix is merged.

I am not sure that:
```
@Override  // violation above 'xxx'
void foo() {}
```

will be good for Inputs. Lets allow it only for Examples, where we need compact (by line numbers) java snippets.
